cmd/cli/app/provider/provider_delete.go

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+
+	"github.com/stacklok/minder/internal/util/cli"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+var deleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete a given provider available in a specific project",
+	Long:  `The minder provider delete command deletes a given provider available in a specific project.`,
+	RunE:  cli.GRPCClientWrapRunE(DeleteProviderCommand),
+}
+
+func init() {
+	ProviderCmd.AddCommand(deleteCmd)
+
+	deleteCmd.Flags().StringP("name", "n", "", "Name of the provider to delete")
+	deleteCmd.Flags().StringP("id", "i", "", "ID of the provider to delete")
+	// We allow deleting by name or ID but not both. One of them must be specified.
+	deleteCmd.MarkFlagsMutuallyExclusive("name", "id")
+	deleteCmd.MarkFlagsOneRequired("name", "id")
+}
+
+// DeleteProviderCommand deletes the provider in a specific project
+func DeleteProviderCommand(ctx context.Context, cmd *cobra.Command, conn *grpc.ClientConn) error {
+	client := minderv1.NewProvidersServiceClient(conn)
+
+	project := viper.GetString("project")
+	name := viper.GetString("name")
+	id := viper.GetString("id")
+
```

Comment:
Set `cmd.SilenceUsage = true` here to avoid printing out the command usage if the RPC to the server fails.

See e.g. [line 71 of `provider_enroll.go`](https://github.com/stacklok/minder/blob/main/cmd/cli/app/provider/provider_enroll.go#L71) for an example.